blockchain/thresholdstate.go

```diff
@@ -731,6 +731,67 @@ func (b *BlockChain) IsHeaderCommitmentsAgendaActive(prevHash *chainhash.Hash) (
 	return isActive, err
 }
 
+// isTreasuryAgendaActive returns whether or not the treasury opcodes are
+// enabled from the point of view of the passed block node.
+//
+// It is important to note that, as the variable name indicates, this function
+// expects the block node prior to the block for which the deployment state is
+// desired.  In other words, the returned deployment state is for the block
+// AFTER the passed node.
+//
+// This function MUST be called with the chain state lock held (for writes).
+func (b *BlockChain) isTreasuryAgendaActive(prevNode *blockNode) (bool, error) {
+	const deploymentID = chaincfg.VoteIDTreasury
+	deploymentVer, ok := b.deploymentVers[deploymentID]
+	if !ok {
+		return true, nil
+	}
+
+	state, err := b.deploymentState(prevNode, deploymentVer, deploymentID)
+	if err != nil {
+		return false, err
+	}
+
+	// NOTE: The choice field of the return threshold state is not examined
+	// here because there is only one possible choice that can be active for
+	// the agenda, which is yes, so there is no need to check it.
+	return state.State == ThresholdActive, nil
+}
+
+// isTreasuryAgendaActiveByHash looks up a node by hash and the returns the
+// value of isTreasuryAgendaActive. Not that the dunction returns treasury not
+// enabled if the has is not found.
+//
+// This function MUST be called with the chain state lock held (for reads).
+func (b *BlockChain) isTreasuryAgendaActiveByHash(prevHash *chainhash.Hash) (bool, error) {
+	prevNode := b.index.LookupNode(prevHash)
+	if prevNode == nil {
+		return false, nil // Not found means not active.
+	}
+	return b.isTreasuryAgendaActive(prevNode)
+}
+
+// IsTreasuryAgendaActiveByHash looks up a node by hash and the returns the
+// value of IsTreasuryAgendaActive.
+//
+// This function is safe for concurrent access.
+func (b *BlockChain) IsTreasuryAgendaActiveByHash(prevHash *chainhash.Hash) (bool, error) {
```

Comment:
This should just be called `IsTreasuryAgendaActive` and the existing `IsTreasuryAgendaActive` should be removed.  See my review comment on that one for why.

Reply:
You fixed this.